### DISCUSSION TITLE
Do Deno Cloud services use AWS or its own hardware infrastructure?

### DISCUSSION THREAD

**Discussion by barziahmed** [denoland/deno#22003] [Q&A]
Hello, 

Are Deno Cloud services, like deploy, deno kv, subhosting, etc. built on a cloud provider (aws, gcp, etc) like vercel does, or they use their owned hardware infrastructure? 

If they use their own, how many datacenters they have? and where can I learn more about this topic and specifically the security and privacy side of it.

thanks

**Comment by ajinkyashukla** [comment]
AFAIK, Deno Deploy and Deno KV runs on GCP

**Reply by nberlette** [reply]
Correct on GCP. Additionally, KV supports backups to your own S3 or S3-compatible GCP storage service.
